nexus/reconfigurator/execution/src/omicron_physical_disks.rs

```diff
@@ -0,0 +1,265 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Manges deployment of Omicron physical disks to Sled Agents.
+
+use crate::Sled;
+use anyhow::anyhow;
+use anyhow::Context;
+use futures::stream;
+use futures::StreamExt;
+use nexus_db_queries::context::OpContext;
+use nexus_types::deployment::BlueprintPhysicalDisksConfig;
+use omicron_uuid_kinds::GenericUuid;
+use omicron_uuid_kinds::SledKind;
+use omicron_uuid_kinds::TypedUuid;
+use slog::info;
+use slog::warn;
+use std::collections::BTreeMap;
+use uuid::Uuid;
+
+/// Idempotently ensure that the specified Omicron disks are deployed to the
+/// corresponding sleds
+pub(crate) async fn deploy_disks(
+    opctx: &OpContext,
+    sleds_by_id: &BTreeMap<Uuid, Sled>,
+    sled_configs: &BTreeMap<TypedUuid<SledKind>, BlueprintPhysicalDisksConfig>,
+) -> Result<(), Vec<anyhow::Error>> {
+    let errors: Vec<_> = stream::iter(sled_configs)
```

Comment:
I think this is purely sequential. Do we want to instead make deployment concurrent via `FuturesUnordered` or the like? My main concern is multiple timeouts adding up consecutively, rather than amortized to a single timeout.

Reply:
I think I'd vote for keeping these sequential on this PR, and have a separate issue for blueprint realization concurrency / optimization. I'd maybe also group "error handling" in that (right now, there are several ways an error can prevent further blueprint realization; I think that's probably overly cautious, but relaxing it might be tricky).